require.js

```diff
@@ -8,10 +8,8 @@
   var cache = {};
   var aliases = {};
   var has = ({}).hasOwnProperty;
-
-  var endsWith = function(str, suffix) {
-    return str.indexOf(suffix, str.length - suffix.length) !== -1;
-  };
+  var extExp = /\.[^\/]+$/
+  var indexExp = /\/index(\.[^\/]+)?$/
```

Comment:
Well, we could do the process recursively... meaning this file will get `page.tpl.html`, `page.tpl`, and `page`

However, that would also mean `Object.assign.js`, `Object.assign`, and *!!!* `Object`. Not sure if that's a good thing...

Perhaps only `page.tpl` is ok then? without `page`